### DISCUSSION TITLE
Zigbee2mqtt not starting

### DISCUSSION THREAD

**Discussion by haybolat** [Koenkk/zigbee2mqtt#6382] [Devices]
Hello everyone, I've been working on this for two days. npm start starts in my first setup. does not start at the other time. I get the following error. What do you think I should do? I would be glad if you could help.

[20:20:13] openhabian@RaspbPi4:/opt/zigbee2mqtt$ npm start

> zigbee2mqtt@1.17.1 start /opt/zigbee2mqtt
> node index.js

Zigbee2MQTT:info  2021-02-20 20:20:19: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2021-02-20.20-20-18' filename: log.txt
Zigbee2MQTT:info  2021-02-20 20:20:19: Starting Zigbee2MQTT version 1.17.1 (commit #6484fee)
Zigbee2MQTT:info  2021-02-20 20:20:19: Starting zigbee-herdsman (0.13.59)
Zigbee2MQTT:error 2021-02-20 20:20:19: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-02-20 20:20:19: Failed to start zigbee
Zigbee2MQTT:error 2021-02-20 20:20:19: Exiting...
Zigbee2MQTT:error 2021-02-20 20:20:19: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at Znp.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:136:32)
    at Generator.next (<anonymous>)
    at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:27:71
    at new Promise (<anonymous>)
    at __awaiter (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:23:12)
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:134:49)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:198:14)
    at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:242:12
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.17.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zigbee2mqtt@1.17.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/openhabian/.npm/_logs/2021-02-20T17_20_19_898Z-debug.log

**Comment by Koenkk** [comment]
See: https://www.zigbee2mqtt.io/information/FAQ.html#error-resource-temporarily-unavailable-cannot-lock-port

**Reply by haybolat** [reply]
Hello, I have applied the same as described here.
1)Raspberry Pi users: use a good power supply # I am using an original Raspberry Pi adapter.
2)When you have a SSD connected to the Pi, try connecting the adapter via a powered USB hub. # I do not use SSD. SD card inserted.
3)Make sure the extension cable works # no extension usb cable, Raspberry Pi usb 3.0 is attached directly.
4)For Openhab users: disable zwave binding # I don't use zwave.
5)In case of a CC2530 or CC2531 adapter, verify that don’t have a CC2540 # I'm just using CC2531.


[npmlog.txt](https://github.com/Koenkk/zigbee2mqtt/files/6017498/npmlog.txt)

**Reply by haybolat** [reply]
Hello, will you help?

**Reply by m0wlheld** [reply]
He did, but you didn‘t listen. 

`Zigbee2MQTT:error 2021-02-20 20:20:19: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'`

Something has a lock on your coordinator device, you need to fix that. On RPI, this includes changing cmdline.txt and de-installing / disabling Network-Manager and other tools.

**Reply by haybolat** [reply]
Hello, Thank you for your help. How can I find out what's locked.

cmdline.txt
dwc_otg.lpm_enable=0 console=serial0,115200 console=tty1 root=PARTUUID=8c7237e2-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait

**Reply by m0wlheld** [reply]
Don‘t know. Read the FAQ for hints. Btw, I was on the wrong track, as you obviously an USB device. 

But

> 3)Make sure the extension cable works # no extension usb cable, Raspberry Pi usb 3.0 is attached directly.

You should/must use and extension cable, otherwise you will run into issues while pairing/communicating after you managed to free the adapter. 

USB3 heavily interferes with 2.4GHz networks like Zigbee.

**Reply by haybolat** [reply]
Hello, it gives an error because I am trying to run the automatic application again.

cd / opt / zigbee2mqtt
npm start

**Comment by 22nra** [comment]
I had this exact same problem when "pairing" was enabled using a cc2531 with piZeroW. 
After I disabled pairing have not seen any other issues on startup.
It looks like like pairing puts extra resources on the hardware which affect the startup process.

**Reply by haybolat** [reply]
Hello, I am using Raspberry pi 4. How pairing is canceled. I am a beginner in this matter. I am trying to do it according to the narrated. Can you describe it?

**Reply by 22nra** [reply]
See: https://www.zigbee2mqtt.io/information/configuration.html
This has all the info on all config options available.

**Reply by haybolat** [reply]
Hello, I haven't reached the level of connecting devices yet I've never been able to start it so that I can pair the device.

[myconfiguration.txt](https://github.com/Koenkk/zigbee2mqtt/files/6017522/myconfiguration.txt)

**Reply by 22nra** [reply]
Just a thought but have you flashed the cc2531 firmware? Some 2531's are supplied flashed for z2m and some not. My Pi zero would start z2m from a cold start but if it was restarted while in use (shutdown -r now) it would intermittently fail. Looks like my pairing suggestion is not the correct route and the initial answer from Koenkk gives you the best chance of success :)
